secrets/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// return codes:
+// 1: failed to read secret from env
+// 2: failed to parse hierarchical secret
+// 3: failed to write secret content into file
+func main() {
+	for _, name := range os.Args[1:] {
```

Comment:
Could we move the actual logic out of `main()` so that it can be tested?

Reply:
implemented, and actually found a bug ...